### ISSUE TITLE
new analog option

### ISSUE THREAD

**Issue by grant2258** [libretro/RetroArch#8543]
Well this all works well but how do we test if an input is analog the thing with analog it sends a spike now and then so you can easily tell its not digital. If its on a deazone with no spike there is no way to tell if it is analog or digital. I would suggest returning a none zero number enough to pick up its an analog control. Again its up to you guys

**Comment by rsn8887** [comment]
What you describe is a terrible way to test whether an input is analog or not. I hope it is not being used anywhere. I might be naive, but I would say there should be no need to test whether an input is analog or digital. The API should provide this information to the cores, no testing needed.

**Comment by grant2258** [comment]
Well I not aware of the api providing this information you mind telling me how the api tells us ?

**Comment by grant2258** [comment]
looks like beetle is also checking the value 0. Hopefully there is an api function to tell us this information.

https://github.com/libretro/beetle-psx-libretro/blob/53591985319edc34d83a0858ad9a935b934dcf5c/input.cpp#L208-L238

**Comment by rsn8887** [comment]
The api names have ANALOG in them for all analog inputs.

**Comment by orbea** [comment]
We have an issue template for a reason...

**Comment by grant2258** [comment]
what info do you need ra version is 

[INFO] RetroArch 1.7.6 (Git 8639018976)
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i3-5005U CPU @ 2.00GHz
[INFO] Capabilities: MMX MMXEXT SSE1 SSE2 SSE3 SSSE3 SSE4 SSE4.2 AVX AES
[INFO] Built: Mar 31 2019
[INFO] Version: 1.7.6
[INFO] Git: 8639018976


the question is simple how do you find out if a button is analog or not before starting a core so you can set the controls appropriately. IE a snes pad will be all digital and a xbox pad will have two analog buttons

**Comment by rsn8887** [comment]
Looking at the beetle example more closely, I don't see the problem anymore.

If you want to read analog buttons in the core, use 
```
input_state_cb( player_index, RETRO_DEVICE_ANALOG, RETRO_DEVICE_INDEX_ANALOG_BUTTON, id);
```

If you want to read digital buttons in the core, use something like
```
button = input_state_cb( player_index, RETRO_DEVICE_JOYPAD,0,id );
```
A zero has the same meaning in both cases: not pressed.

Of course, when you read analog buttons, you might not want to treat a 1 as pressed because the user might have set the deadzone to zero, and the values go up to 32k. For digital buttons, I think a 1 should count as pressed.

AFAIK now, the API *does* provide different mechanisms to read analog vs digital. So you know what you are reading.

**Comment by grant2258** [comment]
the thing is this relies on you pressing the button to find out that not ideal for the situation. Here is an example I have a snes pad and a xbox 360 or psx pad. 

I want to setup the controls depending on what pads plugged in. I guess you could do a manual select on options->controls it would be nice to have a way to detect the gamepads capabilities maybe there is i dont know.

**Comment by fr500** [comment]
I don't even understand what's going on...
The new analog deadzone option is a FRONTEND feature and has nothing to do with the API.

**Comment by grant2258** [comment]
I just want to know how to find out if a given controller plugged in has analog axis available  without the need to press every button to find out if it has an analog axis.  

So i can either map the controller as digital or analog.

**Comment by fr500** [comment]
There is not sure way to tell, and the physical gamepad has no bearing in this discussion.
The cores only know about one gamepad. RETROPAD

There are two variations

- RETRO_DEVICE_JOYPAD which looks like a SNES gamepad
- RETRO_DEVICE_ANALOG which looks like a WiiU Pro gamepad/Dualshock

That's all there is to it. Any other variation is up to the frontend

**Comment by grant2258** [comment]
Well that answers the question you cant tell physical gamepads characteristics to map it to an appropiate retropad. I guess ill just add controller types. That aside there is a bug in the deadzone code when you set it the l2and r2 analog triggers stop working for some reason

**Comment by bparker06** [comment]
L2/R2 works fine for me with both analog deadzone/sensitivity set to non-default values, please file a separate bug for this, and make sure you're testing using the latest git master code as well.
